nodejs/week3/meals-sharing/src/backend/api/meals.js

```diff
@@ -56,31 +51,62 @@ router.get("/", (req, res) => {
         res.json(results);
       }
     );
-    //Get meals that has been created after the date
-    //http://localhost:3000/api/meals?createdAfter=2019-12-08
-  } else if (createdAfter) {
+}
+
+function respondWithCreatedAfter(createdAfter) {
+  pool.query(
+    `SELECT * FROM meal WHERE created_date >= '${createdAfter}'`,
+    (error, results, fields) => {
+      if (error) {
+        return res.send(error);
+      }
+      res.json(results);
+    }
+  );
+}
+
+function respondWithLimitMeals(limit) {
+  const limitNumber = parseInt(limit.trim());
     pool.query(
-      `SELECT * FROM meal WHERE created_date >= '${createdAfter}'`,
+      `SELECT * FROM meal LIMIT ${limitNumber}`,
       (error, results, fields) => {
         if (error) {
           return res.send(error);
         }
         res.json(results);
       }
     );
+}
+
+
+router.get("/", (req, res) => {
+  const {
+    maxPrice,
+    availableReservations,
+    title,
+    createdAfter,
+    limit
+  } = req.query;
+  //Get meals that has a price smaller than maxPrice
+  //http://localhost:3000/api/meals?maxPrice=90
+  if (maxPrice) {
+    respondWithMaxPriceLimit(maxPrice);

```

Comment:
YES!!! This is so much nicer to read. And as a programmer if i want to figure out what respondWithMaxPriceLimit does i can just find that function, but i dont need to because it is abstracted away.